doc/Language/syntax.pod6

```diff
@@ -223,109 +223,193 @@ say 'code again';
 
     my $foo:bar<baz> = 'quux';
-    say $foo:bar«baz»; # OUTPUT: «quux␤»
+    say $foo:bar«baz»;                                  # OUTPUT: «quux␤»
     my $take-me:<home> = 'When the glory has no end';
-    say $take-me:['home'];# OUTPUT: «When the glory has no end␤»
+    say $take-me:['home'];                              # OUTPUT: «When the glory has no end␤»
+    my $foo:bar<2> = 5;
+    say $foo:bar(1+1);                                  # OUTPUT: «5␤»
+
+Where an extended identifier comprises two or more colon pairs, their order
+is generally significant:
 
-This extended syntax is used, for intance, for module versions and
-authors in the shape:
-C«HTTP::UserAgent:ver<1.1.16>:auth<github:sergot>».
+    my $a:b<c>:d<e> = 100;
+    my $a:d<e>:b<c> = 200;
+    say $a:b<c>:d<e>;               # OUTPUT: «100␤», NOT: «200␤»
 
-Colonpairs with empty key (e.g. C«:<foo>») aer reserved in subroutine
-names, as they're used to introduce
-L<a new custom operator|/language/functions#Defining_operators> into
-a particular category. Starting with 6.d language, colonpairs with
-C<sym> key (e.g. C«:sym<foo>») are reserved for possible future use.
+An exception to this rule is I<module versioning>; so these identifiers
+effectively name the same module:
 
-There are slight differences between the three constructs: C«<>» cannot be used
-for interpolation of constant names.
+    use ThatModule:auth<Somebody>:ver<2.7.18.28.18>
+    use ThatModule:ver<2.7.18.28.18>:auth<Somebody>
 
-    constant $what= 'are';
+Furthermore, the adverbial pair syntax in extended identifiers supports
+compile-time interpolation, which mandates the use of
+L<constants|/language/terms#Constants> for the interpolation values:
+
+    constant $c = 42;               # Constant binds to Int; $-sigil enables interpolation
+    my $a:foo<42> = "answer";
+    say $a:foo«$c»;                 # OUTPUT: «answer␤»
+
+Although quoting bracketing constructs are generally interchangeable
+in the context of identifiers, they are not identical. In particular,
+angle brackets C«< >» (which mimic single quote interpolation
+characteristics) cannot be used for the interpolation of constant
+names.
+
+    constant $what = 'are';
     my @we:<are>= <the champions>;
-    say @we:«$what»; # «[the champions]␤»
+    say @we:«$what»;                # OUTPUT: «[the champions]␤»
+    say @we:[$what];                # OUTPUT: «[the champions]␤»
+    say @we:<$what>;                # Compilation error: Variable '@we:<$what>' is not declared
+
+=head3 Compound identifiers
+
+A compound identifier is an identifier that is composed of two or more
+ordinary and/or extended identifiers that are separated from one another by a
+double colon C<::>. Note that the C<::> is referred to as a I<double colon>,
+and not as a I<colon pair>. The latter term refers to the adverbial pair syntax
+C<:key<value>> that may be used to create a Pair, or, as described above, to
+create an extended identifier.
```

Comment:
That's an adverbial pair, not a colon pair. Colon pairs include other constructs such as :a.

Reply:
I don't understand what you are saying.

Reply:
Simply say that :: is a double colon. The rest of the paragraph do not help clarify that fact.

Reply:
Done.